sdkproject/Assets/Mapbox/Unity/MeshGeneration/LayerVisualizers/LocationPrefabsLayerVisualizer.cs

```diff
@@ -0,0 +1,309 @@
+			if (cat == LocationPrefabCategories.None || cat == LocationPrefabCategories.AnyCategory)
+			{
+				return containingCategories;
+			}
+
+			//For any other categories other than None and Any
+			foreach(var value in eligibleValues)
+			{
+				var category = (LocationPrefabCategories)value;
+
+				if (category == LocationPrefabCategories.AnyCategory || category==LocationPrefabCategories.None)
+					continue;
+				
+				if((category & cat) != 0) //to check if category is contained in cat
+				{
+					containingCategories.Add(category);
+				}
+			}
+
+			return containingCategories;
+		}
+
+		public override void Create(VectorTileLayer layer, UnityTile tile, Action callback)
+		{
+			//for layers using specific locations, ignore VectorTileLayer and
+			//pass coordinates to the modifierstack.
+			if( (SubLayerProperties as PrefabItemOptions).findByType 
+			   == LocationPrefabFindBy.AddressOrLatLon)
+			{
+				BuildFeatureFromLatLon(layer, tile);
+				if (callback != null)
+				{
+					callback();
+				}
+			}
+			else
+			{
+				base.Create(layer, tile, callback);
+
+			}
+
+		}
+
+		private void BuildFeatureFromLatLon(VectorTileLayer layer, UnityTile tile)
+		{
+			var coordinates = (SubLayerProperties as PrefabItemOptions).coordinates;
+
+			for (int i = 0; i < coordinates.Length; i++)
+			{
+				if(String.IsNullOrEmpty(coordinates[i]))
+				{
+					return;
+				}
+
+				//check if the coordinate is in the tile
+				var coordinate = Conversions.StringToLatLon( coordinates[i] );
+				var coordinateTileId = Conversions.LatitudeLongitudeToTileId(
+					coordinate.x, coordinate.y, tile.InitialZoom);
+
+				if( coordinateTileId.Canonical.Equals(tile.CanonicalTileId))
+				{
+
+					//create new vector feature
+					var feature = new VectorFeatureUnity();
+					feature.Properties = new Dictionary<string, object>();
+					feature.Points = new List<List<Vector3>>();
+
+					//create a point in tilespace from coordinates
+					var latLonPoint = new List<Vector3>();
+
+					//Create a feature point based on the coordinate
+					var _meters = Conversions.LatLonToMeters(coordinate);
+					var _rect = tile.Rect;
+					var _scale = tile.TileScale;
+					var _extent = layer.Extent;
+
+					//vectortile space point (0 - layerExtent)
+					var point = new Vector3((float)((_meters - _rect.Min).x / _rect.Size.x) * _extent, 
+					                        0, 
+					                        (float)((_meters - _rect.Max).y / _rect.Size.y) * _extent
+					                       );
+
+					//UnityTile space
+					latLonPoint.Add(new Vector3((float)(point.x / _extent * _rect.Size.x - (_rect.Size.x / 2)) * tile.TileScale, 
+					                            0, 
+					                            (float)((point.z) / _extent * _rect.Size.y - (_rect.Size.y / 2)) * tile.TileScale
+					                           ));
+
+					//add coordinate feature to feature points
+					feature.Points.Add(latLonPoint);
+
+					//HACK: silence errors when modifiers access feature.Data
```

Comment:
Is this hack still necessary, and if so, can we at least remove the comment that says it's a hack?